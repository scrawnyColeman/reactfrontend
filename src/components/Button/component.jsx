import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { default as MaterialButton } from '@material-ui/core/Button';
import { colours, materialStyles } from '../../constants/styles';

const Button = ({ text, hierarchy = 'primary', size = 'large', onClick }) => {
    const useStyles = makeStyles(materialStyles);
    const classes = useStyles();
    const buttonOverrideStyles = {
        fontSize: size === 'large' ? '1.25rem' : '1rem',
        textTransform: 'unset',
        color: colours.primary,
    };
    return (
        <MaterialButton
            color={hierarchy}
            size={size}
            className={classes.button}
            style={buttonOverrideStyles}
            variant="text"
            onClick={onClick}
        >
            {text}
        </MaterialButton>
    );
};

export default Button;
