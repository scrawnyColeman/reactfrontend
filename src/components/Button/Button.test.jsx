import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React from 'react';
import Button from './component';

Enzyme.configure({ adapter: new Adapter() });

describe('Button component', () => {
    const onClickMock = jest.fn();

    beforeEach(() => {
        onClickMock.mockReset();
    });

    test('primary button matches snapshot', () => {
        const wrapper = shallow(<Button text="primary button" />);
        expect(wrapper).toMatchSnapshot();
    });
    test('secondary button matches snapshot', () => {
        const wrapper = shallow(<Button hierarchy="secondary" text="Second button" />);
        expect(wrapper).toMatchSnapshot();
    });
    test('large button matches snapshot', () => {
        const wrapper = shallow(<Button size="large" text="Second button" />);
        expect(wrapper).toMatchSnapshot();
    });
    test('onClick prop is called when button is clicked', () => {
        const wrapper = shallow(<Button hierarchy="secondary" text="Button" onClick={onClickMock} />);
        wrapper.props().onClick();
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });
});
