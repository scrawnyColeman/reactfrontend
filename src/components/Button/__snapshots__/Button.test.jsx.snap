// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component large button matches snapshot 1`] = `
<WithStyles(ForwardRef(Button))
  color="primary"
  size="large"
  style={
    Object {
      "color": "#15e601",
      "fontSize": "1.25rem",
      "textTransform": "unset",
    }
  }
  variant="text"
>
  Second button
</WithStyles(ForwardRef(Button))>
`;

exports[`Button component primary button matches snapshot 1`] = `
<WithStyles(ForwardRef(Button))
  color="primary"
  size="large"
  style={
    Object {
      "color": "#15e601",
      "fontSize": "1.25rem",
      "textTransform": "unset",
    }
  }
  variant="text"
>
  primary button
</WithStyles(ForwardRef(Button))>
`;

exports[`Button component secondary button matches snapshot 1`] = `
<WithStyles(ForwardRef(Button))
  color="secondary"
  size="large"
  style={
    Object {
      "color": "#15e601",
      "fontSize": "1.25rem",
      "textTransform": "unset",
    }
  }
  variant="text"
>
  Second button
</WithStyles(ForwardRef(Button))>
`;
